toothpick-runtime/src/main/java/toothpick/ScopeImpl.java

```diff
@@ -78,6 +80,7 @@ public ScopeImpl(Object name) {
 
   @Override
   public <T> Lazy<T> getLazy(Class<T> clazz, String name) {
+    crashIfInvalid();
```

Comment:
I don't understand why you mention GC. It looks to me that the chain you describe in: 

> ThreadSafeProviderImpl.get -> InternalProviderImpl.get -> find factory using registry -> factory.createInstance -> return new ClassWithNoDependencies()

is ***not*** using any scope, hence it could be techincally possible to use a lazy after the scope is closed. Why does it matter that the scope is GC'ed or not ?

Also, the chain above looks a bit strange, a scope could not use a factory but a singleton for instance..

But I understand that the scope is not used after the lazy is created. Honestly I think we should not have any different behavior between something provided directly (as an instance) or via a provider or a lazy. I understand that for us it's different, but form a user and from a DI perspective I don't think of any reasonable meaning of having different behaviors. It looks more like an akward implementation detail (we could actually just have called scope.getInstance in the lazy). I can't make any sense out of such a deferred behavior (in time).